### DISCUSSION TITLE
Please help me.Is this a bug or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by Matteo-Barberis** [MyreMylar/pygame_gui#637] [Q&A]
Here is a minimal reproducible example:

The code simply opens and closes an inventory.

PROBLEM 1:

If the inventory_panel is set to visible = 0, like in the example, when the inventory is then visible (as a result of pressing the inventory button), it's not scrollable. The scrollbar doesn't appear (however it should).

PROBLEM 2:

If the inventory_panel is NOT set to visible = 0, but we are simply not defining the visible property, then the inventory will have the scrollbar, but when the panel is set to invisible again, the scrollbar will still be visible.

Is this a bug? Am i doing something wrong? Could anyone help?

`
import pygame
import pygame_gui
import json

pygame.init()

window_size = (1200, 800)
window_width, window_height = window_size
screen = pygame.display.set_mode(window_size)
pygame.display.set_caption("Minimal Inventory Example")

ui_manager = pygame_gui.UIManager(window_size)

inventory_button = pygame_gui.elements.UIButton(
    relative_rect=pygame.Rect((650, 10), (140, 40)),
    text="Inventory",
    manager=ui_manager
)

inventory_panel = pygame_gui.elements.UITextBox(
    relative_rect=pygame.Rect((50, window_height - 500), (window_width - 100, 360)),
    html_text="",
    manager=ui_manager,
    visible = 0
)

clock = pygame.time.Clock()

inventory_visible = False

inventory_data = json.loads("""{
  "inventory": [
    {
      "name": "Small Potion",
      "description": "A small potion that restores a small amount of health.",
      "quantity": 3,
      "type": "consumable",
      "effect": "Restores 20 HP",
      "usable": true,
      "rarity": "common",
      "durability": null,
      "value": 10
    },
    {
      "name": "Small Bomb",
      "description": "A small bomb that when used inflicts damage to entities that are hit by it.",
      "quantity": 2,
      "type": "consumable",
      "effect": "Inflicts 30 damage to HP",
      "usable": true,
      "rarity": "common",
      "durability": null,
      "value": 10
    },
    {
      "name": "Poison Bomb",
      "description": "A small bomb that when used poisons entities hit by it",
      "quantity": 1,
      "type": "consumable",
      "effect": "Inflicts poison damage over time",
      "usable": true,
      "rarity": "common",
      "durability": null,
      "value": 10
    },
    {
      "name": "Whispering Shard",
      "description": "A translucent crystal that flickers with an inner light, said to contain echoes of lost souls.",
      "quantity": 1,
      "type": "Artifact",
      "effect": "Contains whispers of lost souls, can be used for divination.",
      "usable": true,
      "rarity": "Rare",
      "durability": null,
      "value": 250
    },
    {
      "name": "Pair of Ethereal Gloves",
      "description": "Gloves that grant the wearer a fleeting ability to phase through the mist.",
      "quantity": 1,
      "type": "Accessory",
      "effect": "Allows the wearer to phase through misty areas temporarily.",
      "usable": true,
      "rarity": "Uncommon",
      "durability": null,
      "value": 150
    },
    {
      "name": "Misty Essence",
      "description": "A swirling, smoky substance known for its use in enchantments and potions that enhance stealth.",
      "quantity": 5,
      "type": "Component",
      "effect": "Enhances potions and enchantments related to stealth.",
      "usable": false,
      "rarity": "Common",
      "durability": null,
      "value": 30
    }
  ]
}"""
)

def format_inventory(inventory):
    formatted = "<b>Inventory:</b><br>"
    for item in inventory['inventory']:
        formatted += f"<b>{item['name']}</b> (x{item['quantity']})<br>"
        formatted += f"Effects: {item['effect']}<br>"
        formatted += f"Type: {item['type']}, Rarity: {item['rarity']}, Value: {item['value']}<br><br>"
    return formatted

inventory_panel.set_text(format_inventory(inventory_data))

running = True
while running:
    time_delta = clock.tick(60)/1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

        if event.type == pygame.USEREVENT:
            if event.user_type == pygame_gui.UI_BUTTON_PRESSED:
                if event.ui_element == inventory_button:
                    inventory_visible = not inventory_visible
                    inventory_panel.visible = 1 if inventory_visible else 0

        ui_manager.process_events(event)

    ui_manager.update(time_delta)

    screen.fill((255, 255, 255))
    ui_manager.draw_ui(screen)

    pygame.display.update()

pygame.quit()
`

**Comment by AyhemKouki** [comment]
Problem 1: Scrollbar Not Appearing
When you set the visible property of the inventory_panel to 0, it might not correctly initialize the scrollbar's visibility state when you make it visible again . Instead of directly setting the visible property to 0, consider using set_visible() method which is more consistent in updating the visibility of the component.

**Reply by VitoR-InA** [reply]
I have similar problem with UIHorizontalSlider visibility, but your solution didn't work because it doesn't have "set_visible" function: `AttributeError: 'UIHorizontalSlider' object has no attribute 'set_visible'. Did you mean: '_set_visible'?`. What should I do?
